src/manifolds/HyperbolicHyperboloid.jl

```diff
@@ -0,0 +1,291 @@
+    return HyperboloidPoint(p)
+end
+convert(::Type{<:AbstractVector}, p::HyperboloidPoint) = p.value
+
+@doc raw"""
+    convert(::Type{HyperboloidPoint}, p::PoincareBallPoint)
+    convert(::Type{<:AbstractVector}, p::PoincareBallPoint)
+
+convert a point [`PoincareBallPoint`](@ref) `x` (from $ℝ^n$) from the
+Poincaré ball model of the [`Hyperbolic`](@ref) manifold $ℍ^n$ to a [`HyperboloidPoint`](@ref) $π(p) ∈ ℝ^{n+1}$.
+The isometry is defined by
+
+````math
+π(p) = \frac{1}{1-\lVert p \rVert^2}
+\begin{pmatrix}2p_1\\⋮\\2p_n\\1+\lVert p \rVert^2\end{pmatrix}
+````
+
+Note that this is also used, when the type to convert to is a vector.
+"""
+function convert(::Type{HyperboloidPoint}, p::PoincareBallPoint)
+    return HyperboloidPoint(convert(AbstractVector, p))
+end
+function convert(::Type{<:AbstractVector}, p::PoincareBallPoint)
+    return 1 / (1 - norm(p.value)^2) .* [(2 .* p.value)..., 1 + norm(p.value)^2]
+end
+
+@doc raw"""
+    convert(::Type{HyperboloidPoint, p::PoincareHalfSpacePoint)
+    convert(::Type{<:AbstractVector}, x::PoincareHalfSpacePoint)
+
+convert a point [`PoincareHalfSpacePoint`](@ref) `p` (from $ℝ^n$) from the
+Poincaré half plane model of the [`Hyperbolic`](@ref) manifold $ℍ^n$ to a [`HyperboloidPoint`](@ref) $π(p) ∈ ℝ^{n+1}$.
+
+This is done in two steps, namely transforming it to a Poincare ball point and from there further on to a Hyperboloid point.
+"""
+function convert(t::Type{HyperboloidPoint}, p::PoincareHalfSpacePoint)
+    return convert(t, convert(PoincareBallPoint, p))
+end
+function convert(t::Type{<:AbstractVector}, p::PoincareHalfSpacePoint)
+    return convert(t, convert(PoincareBallPoint, p))
+end
+
+@doc raw"""
+    convert(::Type{HyperboloidTVector}, (p,X)::Tuple{PoincareBallPoint,PoincareBallTVector})
+    convert(::Type{<:AbstractVector}, (p,X)::Tuple{PoincareBallPoint,PoincareBallTVector})
+
+Convert the [`PoincareBallTVector`](@ref) `X` from the tangent space at `p` to a
+[`HyperboloidTVector`](@ref) by computing the push forward of the isometric map, cf.
+[`convert(::Type{HyperboloidPoint}, p::PoincareBallPoint)`](@ref).
+
+The push forward $π_*(p)$ maps from $ℝ^n$ to a subspace of $ℝ^{n+1}$, the formula reads
+
+````math
+π_*(p)[X] = \begin{pmatrix}
+    \frac{2X_1}{1-\lVert p \rVert^2} + \frac{4}{(1-\lVert p \rVert^2)^2}⟨X,p⟩p_1\\
+    ⋮\\
+    \frac{2X_n}{1-\lVert p \rVert^2} + \frac{4}{(1-\lVert p \rVert^2)^2}⟨X,p⟩p_n\\
+    \frac{4}{(1-\lVert p \rVert^2)^2}⟨X,p⟩
+\end{pmatrix}.
+````
+"""
+function convert(
+    ::Type{HyperboloidTVector},
+    (p, X)::Tuple{PoincareBallPoint,PoincareBallTVector},
+)
+    return HyperboloidTVector(convert(AbstractVector, (p, X)))
+end
+function convert(
+    ::Type{T},
+    (p, X)::Tuple{PoincareBallPoint,PoincareBallTVector},
+) where {T<:AbstractVector}
+    t = (1 - norm(p.value)^2)
+    den = 4 * dot(p.value, X.value) / (t^2)
+    c1 = (2 / t) .* X.value + den .* p.value
+    return [c1..., den]
```

Comment:
```suggestion
    return vcat(c1, den)
```
This is a more standard way of doing this.